sdk/endpoint/neighbor.go

```diff
@@ -0,0 +1,65 @@
+package endpoint
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/connectioncontext"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/sirupsen/logrus"
+	"net"
+)
+
+type NeighborEndpoint struct {
+	BaseCompositeEndpoint
+}
+
+func (ne *NeighborEndpoint) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+	if ne.GetNext() == nil {
+		err := fmt.Errorf("Neighbor endpoint needs next")
+		logrus.Errorf("%v", err)
+		return nil, err
+	}
+
+	newConnection, err := ne.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	addrs, err := net.Interfaces()
```

Comment:
I think this is useless to have, since this is just was used as example for capturing list of arp entries, I think better to provide function for users or list of arp entries to pass.

Reply:
It seems like we need more real-world scenarios for neighbors endpoint, because now we don't know how it will be used. So I'd prefer to leave endpoint as it is until use cases become clearer. The main idea of that PR - exclude setting neighbors from widely used IPAM endpoint.

Reply:
I can agree to keep the code as it is here. It can be used as an example and if we need we can move improve it in a follow-up PR.

Reply:
SDK should not have any logic this way, since it was written only to check ARP entries and i'm mostly sure any real NSE will not do it this way :-)